lib/types.ts

```diff
@@ -1,17 +1,7 @@
 /** Essentials */
 export type Primitive = string | number | boolean | bigint | symbol | undefined | null;
 export type Builtin = Primitive | Function | Date | Error | RegExp;
-export type IsTuple<T> = T extends [infer A]
-  ? T
-  : T extends [infer A, infer B]
-  ? T
-  : T extends [infer A, infer B, infer C]
-  ? T
-  : T extends [infer A, infer B, infer C, infer D]
-  ? T
-  : T extends [infer A, infer B, infer C, infer D, infer E]
-  ? T
-  : never;
+export type IsTuple<T> = T extends any[] ? (any[] extends T ? never : T) : never;
```

Comment:
I found that check based on my article – https://blog.beraliv.dev/2021-03-26-typed-get/#33-tuples

I built the table for that and distinguished the cases where array and where tuple